### ISSUE TITLE
error MSB4066: The attribute "Text" in element <Message> is unrecognized.

### ISSUE THREAD

**Issue by pchung13** [MicrosoftDocs/visualstudio-docs#5666]
I followed the example in trying to set up a message task in my build, but I am getting this error. Is Message no longer a supported way of logging in your MSBuild files?

**Comment by pchung13** [comment]
⨯ pachun@PACHUN-DEVBOX1  D:\Fx\Main   FixVersioning ≢ +1 ~2 -0 !                                                                                                                                                       [17:21]
❯ msbuild /m
Microsoft (R) Build Engine version 16.6.0+5ff7b0c9e for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 8/3/2020 5:21:31 PM.
     1>Project "D:\Fx\Main\dirs.proj" on node 1 (default targets).
     1>D:\Fx\Main\GlobalImports.props(10,14): error MSB4066: The attribute "Text" in element <Message> is unrecognized. [D:\Fx\Main\dirs.proj]
     1>Done Building Project "D:\Fx\Main\dirs.proj" (default targets) -- FAILED.

Build FAILED.

       "D:\Fx\Main\dirs.proj" (default target) (1) ->
         D:\Fx\Main\GlobalImports.props(10,14): error MSB4066: The attribute "Text" in element <Message> is unrecognized. [D:\Fx\Main\dirs.proj]

    0 Warning(s)
    1 Error(s)

**Comment by msebolt** [comment]
@pchung13  Thank you for the feedback.

Please include a reference to the article in question and we can assign this to the author for further review.

**Comment by pchung13** [comment]
@msebolt https://docs.microsoft.com/en-us/visualstudio/msbuild/message-task?view=vs-2019

**Comment by msebolt** [comment]
Great, thank you.  Assigning this to the author for further review.

The issues has also been reported [here](https://stackoverflow.com/questions/62518031/msbuild-the-attribute-text-in-element-message-is-unrecognized).

Perhaps try a hard-coded string, such as [this example](https://stackoverflow.com/questions/7557562/how-do-i-get-the-message-msbuild-task-to-show-up-in-the-visual-studio-project) to see if its caused by unset variables?

**Comment by pchung13** [comment]
Yes, I have also tried using a hard-coded string and it threw the same error as well.

**Comment by ghogen** [comment]
AFAIK there's nothing wrong with Message and the Text attribute, when used as expected as a task in a target. The example from the doc built for me with MSBuild 16.8 Preview 1.

```
c:\Source\Repos\MSBuild\Message-Text>cat test.proj
<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <Target Name="DisplayMessages">
        <Message Text="Project File Name = $(MSBuildProjectFile)" />
        <Message Text="Project Extension = $(MSBuildProjectExtension)" />
    </Target>
</Project>
c:\Source\Repos\MSBuild\Message-Text>msbuild test.proj
Microsoft (R) Build Engine version 16.8.0-preview-20403-05+410928129 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 8/13/2020 5:11:59 PM.
Project "c:\Source\Repos\MSBuild\Message-Text\test.proj" on node 1 (default targets).
DisplayMessages:
  Project File Name = test.proj
  Project Extension = .proj
Done Building Project "c:\Source\Repos\MSBuild\Message-Text\test.proj" (default targets).


Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.56
```

It's possible that you would get an error trying to use Message and Text somewhere other than in a target.  Can you post a small repro case?

**Comment by ghogen** [comment]
@pchung I suspect the issue is the placement of the code in a context where it's not executed. Can you post a small repro case?

**Comment by pchung** [comment]
@ghogen wrong @pchung13 :)
